wit-0.3.0-draft/system-clock.wit

```diff
@@ -36,11 +36,14 @@ interface wall-clock {
     /// [POSIX's Seconds Since the Epoch]: https://pubs.opengroup.org/onlinepubs/9699919799/xrat/V4_xbd_chap04.html#tag_21_04_16
     /// [Unix Time]: https://en.wikipedia.org/wiki/Unix_time
     @since(version = 0.3.0)
-    now: func() -> datetime;
+    now: func() -> instant;
 
     /// Query the resolution of the clock.
     ///
     /// The nanoseconds field of the output is always less than 1000000000.
     @since(version = 0.3.0)
-    resolution: func() -> datetime;
+    resolution: func() -> instant;
+    // NOTE: This return value doesn't represent an exact time, so maybe is not
+    // a correct use of the Instant type. Would it make sense to have a
+    // system-clock::duration analogous to monotonic-clock::duration?
```

Comment:
Yes, I agree there should be a duration type in system-clock, can you add one?

Reply:
Done.

It's a _little_ silly to use a seconds+nanoseconds struct for this given that a u64 of nanoseconds can represent durations up to 585 years and system clock resolution is normally going to be at most 16 milliseconds. Even a u32 of nanoseconds (~4 seconds) would probably suffice. But `system-clock:duration` is the most conceptually correct type and this is not going to be on the critical path anywhere, so it shouldn't really matter. And it does match `clock_getres`.